addons/pos_restaurant/static/src/js/floors.js

```diff
@@ -100,276 +153,208 @@ const PosRestaurantPosGlobalState = (PosGlobalState) => class PosRestaurantPosGl
-        if (!(this.config && this.config.iface_floorplan)) {
-            return orders;
-        } else if (!this.table) {
-            return [];
-        } else {
-            var t_orders = [];
-            for (var i = 0; i < orders.length; i++) {
-                if ( orders[i].table === this.table) {
-                    t_orders.push(orders[i]);
-                }
-            }
-            return t_orders;
-        }
+    getTableOrders(tableId) {
+        return this.get_order_list().filter(order => order.tableId === tableId);
     }
 
-    // get the list of orders associated to a table. FIXME: should be O(1)
-    get_table_orders(table) {
-        var orders   = this.get_order_list();
-        var t_orders = [];
-        for (var i = 0; i < orders.length; i++) {
-            if (orders[i].table === table) {
-                t_orders.push(orders[i]);
-            }
-        }
-        return t_orders;
+    unsetTable() {
+        this._syncTableOrdersToServer();
+        this.table = null;
+        this.set_order(null);
     }
 
-    // get customer count at table
-    get_customer_count(table) {
-        var orders = this.get_table_orders(table).filter(order => !order.finalized);
-        var count  = 0;
-        for (var i = 0; i < orders.length; i++) {
-            count += orders[i].get_customer_count();
-        }
-        return count;
+    setCurrentOrderToTransfer() {
+        this.orderToTransfer = this.selectedOrder;
     }
 
-    // When we validate an order we go back to the floor plan.
-    // When we cancel an order and there is multiple orders
-    // on the table, stay on the table.
-    on_removed_order(removed_order,index,reason){
-        if (this.config.iface_floorplan) {
-            var order_list = this.get_order_list();
-            if (reason === 'abandon') {
-                this.db.set_order_to_remove_from_server(removed_order);
-            }
-            if( (reason === 'abandon' || removed_order.temporary) && order_list.length > 0){
-                this.set_order(order_list[index] || order_list[order_list.length - 1], { silent: true });
-            } else if (order_list.length === 0) {
-                this.table ? this.set_order(null) : this.set_table(null);
-            }
-        } else {
-            super.on_removed_order(...arguments);
+    async transferTable(table) {
+        this.table = table;
+        try {
+            await this._syncTableOrdersFromServer(table.id);
+        } catch (error) {
+            throw error;
+        } finally {
+            this.orderToTransfer.tableId = table.id;
+            this.set_order(this.orderToTransfer);
+            this.orderToTransfer = null;
         }
     }
+    getCustomerCount(tableId) {
+        const tableOrders = this.getTableOrders(tableId).filter(order => !order.finalized);
+        let count = 0;
+        tableOrders.forEach(order => count+= order.getCustomerCount());
+        return count;
```

Comment:
I forgot about this method. 
Don't forget the initial value in case the array can be empty, otherwise it'll throw an error :smile: 
```suggestion
        return tableOrders.reduce((count, order) => count + order.getCustomerCount(), 0);
```

Reply:
`[].reduce((acc, item) => acc + item, 0)` won't throw an error, so I think it's safe to use reduce. Unless `tableOrders` can be undefined, then calling `reduce` or `forEach` on it will crash.

Reply:
Oh, my bad. True, the initial value. haha